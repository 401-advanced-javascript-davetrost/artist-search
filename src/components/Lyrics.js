import React from 'react';
import { useParams } from 'react-router-dom';
import { useLyrics } from '../containers/use-lyrics';
import styles from './Lyrics.css';

export default function Lyrics() {
  const { artist, album, track } = useParams();
  const { lyrics } = useLyrics(artist, track);

  return (
    <>
      <div className={styles.Lyrics}>
        <h1>Lyrics for {track}</h1>
        <h2>by {artist} on the album {album}</h2>
        <pre>{lyrics}</pre>
        {!lyrics &&
          <>
            <p>Sorry, no lyrics found!</p>
            <img src="/src/assets/giphy.gif" />
          </>}
      </div>
    </>
  );
}
