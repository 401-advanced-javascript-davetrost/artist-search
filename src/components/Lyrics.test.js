import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

import reactRouterMock from 'react-router-dom';
jest.mock('react-router-dom', () => ({}));

import useLyricsMock from '../containers/use-lyrics.js';
jest.mock('../containers/use-lyrics.js', () => ({}));

reactRouterMock.useParams = () => ({ artist: 'Tribe Called Qwest', track: 'phony rappers' });

describe('Lyrics component', () => {
  
  it('renders Loading', () => {
    useLyricsMock.useLyrics = () => ({ 
      loading: true 
    });
    const wrapper = shallow(<Lyrics />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders Lyrics', () => {
    useLyricsMock.useLyrics = () => ({ 
      loading: false, 
      lyrics: 'Phony rappers. Who do not write. Phony rappers. Who do not excite. Phony rappers. Check it out, aight' 
    });
    const wrapper = shallow(<Lyrics />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Lyrics not found', () => {
    useLyricsMock.useLyrics = () => ({ 
      loading: false,
      lyrics: ''
    });
    const wrapper = shallow(<Lyrics />);
    expect(wrapper).toMatchSnapshot();
  });


});
