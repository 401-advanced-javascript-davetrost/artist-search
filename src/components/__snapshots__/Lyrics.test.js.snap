// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics component renders Loading 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;

exports[`Lyrics component renders Lyrics 1`] = `
<Fragment>
  <div
    className="Lyrics"
  >
    <h1>
      phony rappers
    </h1>
    <h2>
      by: 
      Tribe Called Qwest
    </h2>
    <pre>
      Phony rappers. Who do not write. Phony rappers. Who do not excite. Phony rappers. Check it out, aight
    </pre>
  </div>
</Fragment>
`;

exports[`Lyrics component renders Lyrics not found 1`] = `
<Fragment>
  <div
    className="Lyrics"
  >
    <h1>
      Sorry, no lyrics found!
    </h1>
    <img
      src="/src/assets/giphy.gif"
    />
  </div>
</Fragment>
`;
